InterfaceStubGenerator.Shim/ModuleInitializer.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Refit.Generator
+{
+    internal static class ModuleInitializer
+    {
+        internal static void Initialize()
+        {
+            try
+            {
+                var codeAnalysis = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(
+                    assembly => assembly.GetName().Name == "Microsoft.CodeAnalysis");
+                var codeAnalysisVersion = codeAnalysis.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version;
+                if (!Version.TryParse(codeAnalysisVersion, out var version))
+                {
+                    return;
+                }
+
+                var implementationName = version switch
+                {
+                    { Major: >= 4 } => "InterfaceStubGenerator.Roslyn40.dll",
+                    _ => "InterfaceStubGenerator.Roslyn38.dll",
```

Comment:
📝 The implementations all have different embedded resource names so they can be loaded by name, even though their assembly identities are the same.